src/System.Data.DataSetExtensions/tests/System/Data/DataRowComparerTests.cs

```diff
@@ -0,0 +1,378 @@
+                true
+            };
+
+            yield return new object[]
+            {
+                ArrayRow(new long[] { 1, 2, 3 }),
+                ArrayRow(new long[] { 1, 2, 4 }),
+                false
+            };
+
+            yield return new object[]
+            {
+                ArrayRow(new long[] { 1, 2, 3 }),
+                ArrayRow(new int[] { 1, 2, 3 }),
+                false
+            };
+
+            yield return new object[]
+            {
+                ArrayRow(new int[] { 1, 2, 3 }),
+                ArrayRow(new int[] { 1, 2 }),
+                false
+            };
+
+            yield return new object[]
+            {
+                ArrayRow(new int[] { 1, 2, 3 }),
+                ArrayRow(new int[2,2]),
+                false
+            };
+
+            yield return new object[]
+            {
+                ArrayRow(new int[2,2]),
+                ArrayRow(new int[2,2]),
+                false
+            };
+
+            yield return new object[]
+            {
+                ArrayRow(new int[2]),
+                ArrayRow(null),
+                false
+            };
+
+            // Different count.
+            yield return new object[]
+            {
+                table1.Rows.Add(1, 2, null),
+                table2.Rows.Add(1),
+                false
+            };
+
+            // Null argument.
+            yield return new object[]
+            {
+                table1.Rows.Add(1, 2, null),
+                null,
+                false
+            };
+
+            yield return new object[]
+            {
+                null,
+                table1.Rows.Add(1, 2, null),
+                false
+            };
+
+            yield return new object[]
+            {
+                null,
+                null,
+                true
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(Equals_TestData))]
+        public void Equals_Rows_ReturnsExpected(DataRow row1, DataRow row2, bool expected)
+        {
+            Assert.Equal(expected, DataRowComparer.Default.Equals(row1, row2));
+            Assert.Equal(expected, DataRowComparer.Default.Equals(row2, row1));
+        }
+
+        [Fact]
+        public void Equals_NullStringValueInStringArrayInRightColumn_ThrowsNullReferenceException()
+        {
+            var table = new DataTable("Table");
+            DataColumn column = table.Columns.Add("Column");
+            column.DataType = typeof(Array);
+
+            DataRow row1 = table.Rows.Add(new object[] { new string[] { null } });
+            DataRow row2 = table.Rows.Add(new object[] { new string[] { null } });
+            DataRow row3 = table.Rows.Add(new object[] { new string[] { "abc" } });
+
+            Assert.Throws<NullReferenceException>(() => DataRowComparer.Default.Equals(row1, row2));
+            Assert.Throws<NullReferenceException>(() => DataRowComparer.Default.Equals(row2, row1));
```

Comment:
Is this assert supposed to test if row 2 and row 3 are equal? The first assert already looks for row 1 and row 2 equivalence